Source/utils/endian_int.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+#include <type_traits>
+
+namespace devilution {
+
+template<typename T>
+T SwapLE(T in)
+{
+	static_assert(std::is_integral_v<T> || std::is_enum_v<T>);
+	static_assert((sizeof(T) == 1) || (sizeof(T) == 2) || (sizeof(T) == 4) || (sizeof(T) == 8));
```

Comment:
`static_assert` requires a message on older compilers

```suggestion
	static_assert(std::is_integral_v<T> || std::is_enum_v<T>, "unsupported type");
	static_assert((sizeof(T) == 1) || (sizeof(T) == 2) || (sizeof(T) == 4) || (sizeof(T) == 8), "unsupported type");
```